cli/src/pcluster/resources/custom_resources/custom_resources_code/utils/ec2_utils.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
+#  the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List
+
+import boto3
+from utils.retry_utils import retry
+
+
+@retry(max_retries=5, wait_time_seconds=3)
+def list_cluster_instance_ids_iterator(
+    cluster_name: str,
+    instance_state: List[str] = ("pending", "running", "stopping", "stopped"),
+    node_type: List[str] = None,
+    batch_size: int = 100,
```

Comment:
I'd say we should increase the batch_size. What about 1000? or 500?

Reply:
Let's go with 500